students/ian_letourneau/Lesson06/mailroom.py

```diff
@@ -0,0 +1,169 @@
+        total = 0
+        donations = donors[donor]
+        for value in donations:
+            total += value
+        donorFile = donor.replace(" ", "")
+        # Format .txt filename using the donors name and the current date. Then
+        # create file and write thank you letter within open .txt file.
+        filename = "{}_{}_{}_{}.txt".format(
+            donorFile, now.year, now.month, now.day)
+        f = open(filename, "w+")
+        f.write('Dear {}, we wanted to say thank you for your generous total'
+                ' donations of ${:.2f}! We hope to continue seeing you in'
+                ' the future!'.format(
+                    donor, float(total)))
+        f.close()
+    print("Letters have been sent to all donors.")
+    return "Letters have been sent to all donors."
+
+
+if __name__ == '__main__':
+    """This area is always exectued by the script. It contains the main menu
+    prompt as well as the calls to the previous two functions depending on which
+    option was chosen. The third option allows the user to quit the script."""
+    response = ''
+    # Prompt menu and repeat until the "Quit" option has been chosen
+    while response != '4':
+        print("""\nHello User! What would you like to do?
+1) Send a Thank You card
+2) Create a report of donors
+3) Send a Thank You to everyone
+4) Quit \n""")
+        response = (
+            input("Please input a numerical value for"
+                  " what you would like to do: "))
+        while response not in ('1', '2', '3', '4'):
+            response = (
+                input("Invalid input. Please input a numerical"
+                      " value for what you would like to do: "))
+
+        # Once response has been verified and is not 3 to quit, call the
+        # appropriate function
+        if response == '1':
+            print("")
+            name_prompt = ("Please enter a full name to send to or enter"
+                           " 'list' for a full list of donors('q' to quit"
+                           " to menu): ")
+            name = input(name_prompt)
+            if name == 'q':
+                continue
+            # If list is chosen, print a list of donors and reprompt for a
+            #  name, 'q' # to quit to main menu
+            while name == 'list':
+                for donor in donors:
+                    print(donor)
+                name = input(name_prompt)
+            if name == 'q':
+                continue
+            # Once name is input, prompt for a donation amount, 'q' to quit
+            amount = input(
+                "Please enter the amount that was donated('q' to quit"
+                " to menu): ")
+            if amount == 'q':
+                continue
+            # Call the send Thank You latter function passing user input
+            send_thank_you(name, amount)
+        elif response == '2':
+            create_report()
+        elif response == '3':
+            send_all()
```

Comment:
you should be using a dict switch to navigate to main functions (that was part of Lesson 4).

example:
```
main_menu = {"1": thank_you, "2": create_report, "3": exit_menu}
main_menu[response]()
```